src/queue.rs

```diff
@@ -18,7 +19,7 @@ pub use self::extras::Extras;
 use std::time::Instant;
 
 /// A cell containing a task and needed extra information.
-pub trait TaskCell {
+pub trait TaskCell: 'static {
```

Comment:
The skiplist's value must be 'static. I think it's fair that a task spawned in the thread pool should live long enough.

Reply:
I don't see why skiplist's value should be 'static. Technically, as long as the value outlive skiplist, it should be OK. If that's an API shortcoming, then the constraint should be put into priority queue instead of here.

Reply:
It looks like an API shortcoming. I tried making the constraint priority-queue-only, but because we use `enum` to dispatch instead of trait, the constraint propagates up.